### ISSUE TITLE
Exception while invoking method 'MeteorToys_impersonate_account' 

### ISSUE THREAD

**Issue by miri-am** [MeteorToys/allthings#26]
```
I20150812-13:11:02.859(2)? Exception while invoking method 'MeteorToys_impersonate_account' TypeError: Cannot read property 'username' of null
I20150812-13:11:02.860(2)?     at [object Object].Meteor.(anonymous function).MeteorToys_impersonate_account (packages/meteortoys:authenticate/server/methods.js:1:1)
I20150812-13:11:02.860(2)?     at [object Object].methodMap.(anonymous function) (packages/meteorhacks:kadira/lib/hijack/wrap_session.js:164:1)
I20150812-13:11:02.860(2)?     at maybeAuditArgumentChecks (packages/ddp/livedata_server.js:1617:1)
I20150812-13:11:02.860(2)?     at packages/ddp/livedata_server.js:648:1
I20150812-13:11:02.860(2)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20150812-13:11:02.860(2)?     at packages/ddp/livedata_server.js:647:1
I20150812-13:11:02.861(2)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20150812-13:11:02.861(2)?     at [object Object]._.extend.protocol_handlers.method (packages/ddp/livedata_server.js:646:1)
I20150812-13:11:02.861(2)?     at packages/meteorhacks:kadira/lib/hijack/wrap_session.js:52:1
I20150812-13:11:02.861(2)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
```

**Comment by abernix** [comment]
I also receive this message, but only in my jasmine testing environment in my client integration logs.   I'm not sure exactly what is causing it.  I just started using it, so this might be really old.

**Comment by came** [comment]
same here
=> App running at: http://localhost:3000/
I20150925-16:05:23.150(2)? Exception while invoking method 'MeteorToys_impersonate_account' TypeError: Cannot read property 'username' of null
I20150925-16:05:23.150(2)?     at [object Object].Meteor.(anonymous function).MeteorToys_impersonate_account (packages/meteortoys:authenticate/server/methods.js:1:1)
I20150925-16:05:23.150(2)?     at [object Object].methodMap.(anonymous function) (packages/meteorhacks:kadira/lib/hijack/wrap_session.js:164:1)
I20150925-16:05:23.150(2)?     at maybeAuditArgumentChecks (packages/ddp/livedata_server.js:1617:1)
I20150925-16:05:23.150(2)?     at packages/ddp/livedata_server.js:648:1
I20150925-16:05:23.150(2)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20150925-16:05:23.150(2)?     at packages/ddp/livedata_server.js:647:1
I20150925-16:05:23.151(2)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20150925-16:05:23.151(2)?     at [object Object]._.extend.protocol_handlers.method (packages/ddp/livedata_server.js:646:1)
I20150925-16:05:23.151(2)?     at packages/meteorhacks:kadira/lib/hijack/wrap_session.js:52:1
I20150925-16:05:23.151(2)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)

**Comment by abernix** [comment]
Still getting this with the latest version.

**Comment by radupopa2010** [comment]
SAME HERE:

I20151009-14:01:31.433(-7)? Exception while invoking method 'MeteorToys' TypeError: Cannot read property 'ToyKit' of undefined
I20151009-14:01:31.463(-7)?     at [object Object].Meteor.(anonymous function).MeteorToys (packages/msavin_mongol/server/methods.js:1:1)
I20151009-14:01:31.478(-7)?     at maybeAuditArgumentChecks (livedata_server.js:1692:12)
I20151009-14:01:31.481(-7)?     at livedata_server.js:708:19
I20151009-14:01:31.484(-7)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20151009-14:01:31.486(-7)?     at livedata_server.js:706:40
I20151009-14:01:31.490(-7)?     at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
I20151009-14:01:31.493(-7)?     at livedata_server.js:704:46
I20151009-14:01:31.495(-7)?     at tryCallTwo (/home/ubuntu/.meteor/packages/promise/.0.5.0.th99ec++os+web.browser+web.cordova/npm/node_modules/meteor-promise/node_modules/promise/lib/core.js:45:5)
I20151009-14:01:31.499(-7)?     at doResolve (/home/ubuntu/.meteor/packages/promise/.0.5.0.th99ec++os+web.browser+web.cordova/npm/node_modules/meteor-promise/node_modules/promise/lib/core.js:171:13)
I20151009-14:01:31.501(-7)?     at new Promise (/home/ubuntu/.meteor/packages/promise/.0.5.0.th99ec++os+web.browser+web.cordova/npm/node_modules/meteor-promise/node_modules/promise/lib/core.js:65:3)

**Comment by msavin** [comment]
Guessing you are getting that in the terminal? I thought I patched it but perhaps not. Do the toys still work?

**Comment by radupopa2010** [comment]
I am running meteor on Ubuntu Server 14.04 The error appears after I refresh the app in the web browser.
Don't know how to check if Toys are working...This is my first app.It was working fine before updating   
Best Regards,

## Radu Gabriel Popa

## E-mail: radupopa2010@yahoo.comMobile: (+1) 619 9404 8022

```
  From: Max Savin <notifications@github.com>
```

 To: MeteorToys/allthings allthings@noreply.github.com 
Cc: radupopa2010 radupopa2010@yahoo.com 
 Sent: Friday, October 9, 2015 2:16 PM
 Subject: Re: [allthings] Exception while invoking method 'MeteorToys_impersonate_account' (#26)

Guessing you are getting that in the terminal? I thought I patched it but perhaps not. Do the toys still work?

**Comment by abernix** [comment]
I should have clarified – though I don't think I even realized it was different.  Right now, I'm getting this in the **Browser** console, although I think I used to get it in my Velocity server logs. (???)  Not entirely sure – maybe I never paid attention. :smirk:

There was a similar browser issue a while back – perhaps what you were thinking of – msavin/Mongol#97, or maybe somehow the error just got moved from server to client. :confused:

**Comment by radupopa2010** [comment]
So after I uninstalled mongol there were no errors, so it seams it needs a new patch :)   Best Regards,

## Radu Gabriel Popa

## E-mail: radupopa2010@yahoo.comMobile: (+1) 619 9404 8022

```
  From: Max Savin <notifications@github.com>
```

 To: MeteorToys/allthings allthings@noreply.github.com 
Cc: radupopa2010 radupopa2010@yahoo.com 
 Sent: Friday, October 9, 2015 2:16 PM
 Subject: Re: [allthings] Exception while invoking method 'MeteorToys_impersonate_account' (#26)

Guessing you are getting that in the terminal? I thought I patched it but perhaps not. Do the toys still work?

**Comment by marbemac** [comment]
Seeing this as well, have to remove meteor toys for now :(.

```
Exception while simulating the effect of invoking 'MeteorToys_impersonate_account' TypeError: Cannot read property 'username' of null
    at Meteor.(anonymous function).MeteorToys_impersonate_account (http://localhost:3001/packages/meteortoys_authenticate.js?ef8e533354e0e765bd32b547109917e5c4958035:141:1388)
    at http://localhost:3001/packages/ddp-client.js?250b63e6c919c5383a0511ee4efbf42bb70a650f:3911:25
    at _.extend.withValue (http://localhost:3001/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:971:17)
    at _.extend.apply (http://localhost:3001/packages/ddp-client.js?250b63e6c919c5383a0511ee4efbf42bb70a650f:3902:54)
    at _.extend.call (http://localhost:3001/packages/ddp-client.js?250b63e6c919c5383a0511ee4efbf42bb70a650f:3780:17)
    at http://localhost:3001/packages/meteortoys_authenticate.js?ef8e533354e0e765bd32b547109917e5c4958035:123:2207 TypeError: Cannot read property 'username' of null
    at Meteor.(anonymous function).MeteorToys_impersonate_account (http://localhost:3001/packages/meteortoys_authenticate.js?ef8e533354e0e765bd32b547109917e5c4958035:141:1388)
    at http://localhost:3001/packages/ddp-client.js?250b63e6c919c5383a0511ee4efbf42bb70a650f:3911:25
    at _.extend.withValue (http://localhost:3001/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:971:17)
    at _.extend.apply (http://localhost:3001/packages/ddp-client.js?250b63e6c919c5383a0511ee4efbf42bb70a650f:3902:54)
    at _.extend.call (http://localhost:3001/packages/ddp-client.js?250b63e6c919c5383a0511ee4efbf42bb70a650f:3780:17)
    at http://localhost:3001/packages/meteortoys_authenticate.js?ef8e533354e0e765bd32b547109917e5c4958035:123:2207
```

**Comment by msavin** [comment]
So - the log errors should be fixed in the new 2.2.0 version (if not in an earlier version). I've tested everything and no signs of it. To update, remove your Meteor Toys package and run `meteor add meteortoys:allthings@2.2.0`. If this issue repeats, or you have other issues, please let me know.

@marbemac it sounds like you are having an unrelated issue, can you please update and then submit a new ticket if it persists.
